packages/fuels-core/src/types/tx_status.rs

```diff
@@ -170,6 +170,13 @@ impl TxStatus {
             _ => vec![],
         }
     }
+
+    pub fn is_final(&self) -> bool {
+        matches!(
+            self,
+            TxStatus::Success(_) | TxStatus::Failure(_) | TxStatus::SqueezedOut(_)
+        )
+    }
```

Comment:
If we do all above I would not add this. Less code to maintain :)
```suggestion
```

Reply:
Still needed until https://github.com/FuelLabs/fuels-rs/issues/1647